### ISSUE TITLE
Comments crossing a buffer block not handled correctly

### ISSUE THREAD

**Issue by JimF42** [JoshClose/CsvHelper#444]
I have a large csv file to process that has a few comments in it. As it happens during my testing one of the comments happens to cross a buffer block boundary and is not being handled correctly.

AppendField is being called and placing the first part of the comment into `field` that later is used for the next record.

I have made a change to the source that seems to suit me, but I am not sure how robust the change is so I am leery about posting a pull request, plus the fact that I have had to tweak the code to handle less than perfect CSV files as I don't have the luxury of demanding perfectly compliant CSV files and I must take what is given to me.

Here is the change that I have applied to my code for the comment bug. I added 1&& !inComment` to the code snip below in CsvParser.cs.

```c#
protected bool GetChar(out char ch, ref int fieldStartPosition, ref int rawFieldStartPosition, ref string field, ref bool fieldIsBad, bool prevCharWasDelimiter, ref int recordPosition, ref int fieldLength, bool inComment, bool inDelimiter, bool inQuotes, bool isPeek)
        {
            if (readerBufferPosition == charsRead)
            {
                // We need to read more of the stream.

                if (!inDelimiter && !inComment) // added "&& !inComment"
                {
                    // The buffer ran out. Take the current
                    // text and add it to the field.
                    AppendField(ref field, fieldStartPosition, fieldLength);
                }
```

**Comment by JoshClose** [comment]
If you can make a failing unit test that shows the issue that passes after your change, and all the other unit tests still pass, make a pull request.

**Comment by JimF42** [comment]
Because of work deadlines and since I have to use a modified version of the code for other reasons (quotes be escaped via \" and spaces around delimiters), that will probably not happen anytime soon by me--hopefully someone else will.

**Comment by JoshClose** [comment]
No worries. I won't be able to get to it for a while either. :(

**Comment by ChangePlaces** [comment]
Having exactly the same problem and can confirm the change of adding ` && !inComment` solves the problem.

**Comment by JoshClose** [comment]
I pushed the fix for this in version 2.13.4 to NuGet.
